docs/ref/models/querysets.txt

```diff
@@ -1863,16 +1863,19 @@ Returns a tuple of ``(object, created)``, where ``object`` is the retrieved or
 created object and ``created`` is a boolean specifying whether a new object was
 created.
 
-This is meant as a shortcut to boilerplatish code. For example::
+This is meant as a shortcut to boilerplatish code and to prevent duplicate objects
+from being created when requests are made in parallel. For example::
 
     try:
         obj = Person.objects.get(first_name='John', last_name='Lennon')
     except Person.DoesNotExist:
         obj = Person(first_name='John', last_name='Lennon', birthday=date(1940, 10, 9))
         obj.save()
 
-This pattern gets quite unwieldy as the number of fields in a model goes up.
-The above example can be rewritten using ``get_or_create()`` like so::
+This pattern gets quite unwieldy as the number of fields in a model goes up,
+and if multiple requests are made concurrently, two ``Person`` objects with the
+same name may be created. The above example can be rewritten using
```

Comment:
This first sentence could be cut in two (it's quite complex). Also I think the concurrency issue is more important than the unwieldiness, so it would be better to present that idea first?

Reply:
Is there any reason to bring up the unwieldiness of the try/except version? The user can presumably see that for themselves. It seems like it's trying to justify the existence of `get_or_create`, which is already justified on other grounds.

It might also be worth changing the order of the clauses in the first sentence of the previous paragraph, from
> This is meant as a shortcut to boilerplatish code and to prevent duplicate objects
from being created when requests are made in parallel.

to
> This is meant to prevent duplicate objects from being created when requests are made in parallel, and as a shortcut to boilerplatish code.

Reply:
Both good points! Who cares about unwieldiness when there are data loss bugs?

Reply:
Deleted the clause about unwieldiness and changed the clause order in the previous paragraph as I proposed.